### PR TITLE
Send to server

### PR DESCRIPTION
**Pull request by ababiarz**
client, model: Chat-related models + sending/receiving messages from the global chat

### CODE REVIEW

File: client/src/app/global-chat/global-chat.component.ts [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+import { AuthService } from '../services/auth.service';
 
 @Component({
   selector: 'app-global-chat',
   templateUrl: './global-chat.component.html',
   styleUrls: ['./global-chat.component.scss'],
 })
 export class GlobalChatComponent implements OnInit {
-  messages: string[] = ['message1', 'message2', 'message3'];
+  username = 'benek';
```
Comment (jerry-sky) [inline client/src/app/global-chat/global-chat.component.ts]: Why is there a random hard-coded username?

File: client/src/app/services/auth.service.ts [hunk 2]
```diff
@@ -87,4 +89,25 @@ export class AuthService {
...
+  GetGlobalMessages(): Observable<Message[]> {
+    return this.backend.getGlobalMessages().pipe(
+      map((response) => 
+        response.messages
+      )
+    );
+  }
 }
+
+
```
Comment (jerry-sky) [inline client/src/app/services/auth.service.ts]: Please fix this and other formatting issues (missing newlines at EOF, trailing whitespace and inappropriate line breaking).

File: client/src/app/services/backend.service.ts [hunk 3]
```diff
@@ -16,6 +18,7 @@ export class BackendService {
   private usersUrl = environment.API + '/user';
   private mapUrl = environment.API + '/map';
   private cityUrl = environment.API + '/city';
+  private globalChatUrl = environment.API + '/globalChat';
```
Comment (jerry-sky) [inline client/src/app/services/backend.service.ts]: For API routes and file names we use `caterpillar-case`, not `CamelCase`.

**Review by jerry-sky — CHANGES_REQUESTED** [review]
Please fix these minor issues I have outlined below.

We will merge this branch back to `master` once the backend part of this functionality is ready (@jerry-sky, @Deatomizeer).

**Comment by ababiarz** [comment]
Minor issues related to file formatting and file naming standard fixed.
Added retrieving username from backend.
Added private messages sending.

**Comment by jerry-sky** [comment]
Roger that — I’ll try to implement this in the backend.

@Deatomizeer — we need to implement `model` classes related to chat into SQL tables.

**Comment by jerry-sky** [comment]
Let’s assume that the global chat is chat of `id=1`.

File: server/src/services/chat.service.ts [hunk 4]
```diff
@@ -35,4 +42,91 @@ export class ChatService {
...
+    await this.database.ExecuteInsideDatabaseHarness(async (connection) => {
+      result = await connection.query(
+        'SELECT u1.chatId FROM `' +
+          DatabaseTables.USER_CHAT +
+          '` AS u1 INNER JOIN userChat AS u2 ON u1.chatId = u2.chatId \
+          AND u1.userId = ? AND u2.userId = ? AND u1.chatId <> ?;',
+        [user1.id, user2.id, GLOBAL_CHAT_ID]
+      );
+    });
+    result = JSON.parse(JSON.stringify(result));
```
Comment (jerry-sky) [inline server/src/services/chat.service.ts]: Wow wow wow, back up here a second — if you want to copy an object use `const copy = { ...original }`.

The second thing is that I don’t really understand why would you want to copy and assign the same object… to itself? Am I missing something here? I presume it’s a result of some debugging process :P.

Comment (Deatomizeer) [reply server/src/services/chat.service.ts]: I don't know, it wouldn't work without that 70th line :(

Comment (jerry-sky) [reply server/src/services/chat.service.ts]: Pushed a commit that resolves this issue.

**Review by jerry-sky — APPROVED** [review]
It should be fine!

**Pull request merged**